server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -471,7 +476,10 @@ public class ActionModule extends AbstractModule {
     private final ClusterSettings clusterSettings;
     private final SettingsFilter settingsFilter;
     private final List<ActionPlugin> actionPlugins;
+    // An unmodifiable map containing OpenSearch and Plugin actions
     private final Map<String, ActionHandler<?, ?>> actions;
+    // A dynamic map combining the above immutable actions with dynamic Extension actions
```

Comment:
Moreover a design question. Do we plan to remove the immutable map completely for `actions` in the future and use the mutable one?

Reply:
No, I don't think so.  See my reply to @peternied [here](https://github.com/opensearch-project/OpenSearch/pull/6734#discussion_r1140418316)

Reply:
> I could put everything in a single registry, but this would:

  >  1.  remove the existing @Inject dependency action capability and require updating all the code to fetch from the registry instead
  
   Definitely. As we are moving towards dynamic actions, existing @Inject can be removed.
   >  2. make the currently immutable actions map mutable, removing the ability for developers to rely on actions always being available
   
   OpenSearch actions would be still available because they will be registered during boostrap.
  > 3. probably break all existing plugins that rely on this functionality
  
  Since we are moving towards extensions. Features like this would come more often. We can start a campaign for plugins to migrate to the new change.
  
  All I wanted to say let's not say `No` to deprecating old way of registering actions. Dynamic actions is the way to move forward in the future with plugins migrating to extensions and can have only one map to register actions.

Reply:
> OpenSearch actions would be still available because they will be registered during boostrap.

But they would be allowed to be unregistered.  That's the problem.

We need to treat "same JVM" vs. "remote JVM" actions differently anyway.  The current split does that well.

Reply:
Makes sense. Can we add this detail for having 2 maps in the comment for providing more information?

Reply:
Sure, I'll expand the above comments.